jf_agent/data_manifests/git/adapters/github.py

```diff
@@ -21,15 +21,32 @@ class GithubManifestGenerator(ManifestAdapter):
     '''
 
     def __init__(
-        self, token: str, company: str, org: str, instance: str, verify=True, **kwargs
+        self,
+        token: str,
+        base_url: str,
+        company: str,
+        org: str,
+        instance: str,
+        verify=True,
+        **kwargs,
     ) -> None:
         # Super class fields
         self.company = company
         self.org = org
         self.instance = instance
         # Session fields
         self.token = token
-        self.base_url = 'https://api.github.com/graphql'
+
+        if not base_url:
+            self.base_url = 'https://api.github.com/graphql'
+        elif 'api/v3' in base_url:
+            # Github server can provide an API with a trailing '/api/v3'
+            # replace this with the graphql endpoint
+            self.base_url = base_url.replace('api/v3', 'graphql')
+        else:
+            # Assume all other cases we have a base_url like this:
+            #  https://api.github.com
+            self.base_url = f'{base_url}/graphql'
```

Comment:
Q: Any possibility that the `base_url` would already be `https://api.github.com/graphql`? Since it looks like from line 42 it could already include `api/v3`.

Reply:
Nope, as of right now all of our agent ingest is strictly using the rest API. This `graphql` change is experimental and limited to only manifest generation for the moment